.github/workflows/PR_summary.yml

```diff
@@ -46,10 +46,26 @@ jobs:
         sudo apt-get install -y jq
         # If you have additional dependencies, install them here
 
-    - name: Get changed files
+    - name: Get changed and removed/renamed files
       run: |
         git fetch origin ${{ github.base_ref }}  # fetch the base branch
-        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt  # get the list of changed files
+        # Get the list of all changed files.
+        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt
+        # Get all files which were removed or renamed.
+        echo "Checking for removed files..."
+        git diff --name-only --diff-filter DR origin/${{ github.base_ref }}... | tee removed_renamed_files.txt
+
+    - name: Compute (re)moved files without deprecation
+      run: |
+        touch moved_without_deprecation.txt
+        git checkout ${{ github.base_ref }}
+        for file in $(cat removed_renamed_files.txt) ; do
+          if grep ^deprecated_module $file ; then
+            echo "info: removed file $file contains a deprecation"
+          else
+            echo "**error**: file $file was removed without a module deprecation" >> moved_without_deprecation.txt
```

Comment:
```suggestion
          if grep ^deprecated_module "${file}" ; then
            printf 'info: removed file %s contains a deprecation\n' "${file}"
          else
            printf '**error**: file %s was removed without a module deprecation\n' "${file}" >> moved_without_deprecation.txt
```

Reply:
Reading up on the difference between these, I learned that "echo" in shell scripts is a portability hazard: TIL, thanks!